server/core/src/main/java/org/infinispan/server/core/backup/resources/CacheResource.java

```diff
@@ -115,11 +129,21 @@ public void prepareAndValidateBackup() {
             String zipPath = cacheRoot.resolve(configFile).toString();
             try (InputStream is = zip.getInputStream(zip.getEntry(zipPath))) {
                ConfigurationBuilderHolder builderHolder = parserRegistry.parse(is, null, MediaType.fromExtension(configFile));
-               Configuration cfg = builderHolder.getNamedConfigurationBuilders().get(cacheName).build();
-               log.debugf("Restoring Cache %s: %s", cacheName, cfg.toXMLString(cacheName));
-               // GetOrCreate in the event that a default cache is defined. This also allows cache-configurations to be
-               // modified prior to a restore if only the backed up data is required
-               SecurityActions.getOrCreateCache(cm, cacheName, cfg);
+               Configuration config = builderHolder.getNamedConfigurationBuilders().get(cacheName).build();
+               log.debugf("Restoring Cache %s: %s", cacheName, config.toXMLString(cacheName));
+               Map<String, String> params = new HashMap<>(2);
+               params.put("name", cacheName);
+               params.put("configuration", config.toXMLString(cacheName));
+               TaskContext taskContext = new TaskContext();
+               taskContext.cacheManager(cm);
+               taskContext.parameters(params);
+
+               CompletionStage<?> taskExecution = SecurityActions.executeTask(BackupCreateCacheTask.NAME, taskManager, taskContext);
+               CompletionStages.join(taskExecution);
+
+               // We must define the configuration locally if a zero-capacity node as these are excluded by BackupCreateCacheTask
+               if (cm.getCacheManagerConfiguration().isZeroCapacityNode())
```

Comment:
> I'm not 100% sure it's possible, but personally I would rather use the cluster executor to create the cache locally with `DefaultCacheManager.createCache()` on all the nodes (regardless of capacity) and in parallel also use `DefaultCacheManagerAdmin.getOrCreateCache()` to save the cache config for future joiners.

I'll give it a try.

Reply:
I've made it so that the cluster executor is used to define the configuration on all zero-capacity nodes via `EmbeddedCacheManager#defineConfiguration` and capacity nodes via a local put to the global state cache.